src/main/sensors/battery.c

```diff
@@ -65,6 +69,13 @@ static uint16_t batteryAdcToVoltage(uint16_t src)
 
 static void updateBatteryVoltage(void)
 {
+    #ifdef USE_ESC_TELEMETRY
+    if (batteryConfig->batteryMeterType == BATTERY_SENSOR_ESC && isEscTelemetryActive()) {
+        vbat = getEscTelemetryVbat();
+        return;
```

Comment:
That alone is not going to work, since `vbatLatestADC` contains the raw ADC reading, and vbat is a voltage. Unfortunately this will have to be rewritten to use `vbatLatestMeasurement` as a voltage instead.